File: 2021/01/31/jh20s/contest#225/1737.py
class Solution(object):
    def go1(self,a,b):
        ans = sys.maxint
        A = [ord(i)-ord('a') for i in a]
        B = [ord(i)-ord('a') for i in b]
        for i in range(1,26):
            cnt = 0
            for j in A:
                if j >=i:
                    cnt+=1
            for j in B:
                if j < i:
                    cnt+=1
            ans = min(ans,cnt)
        return ans

    def go2(self,a,b):
        ans = sys.maxint
        A = [ord(i)-ord('a') for i in a]
        B = [ord(i)-ord('a') for i in b]
        for i in range(0,26):
            cnt =0
            for j in A:
                if j!= i:
                    cnt +=1
            for j in B:
                if j!= i:
                    cnt +=1
            ans = min(ans,cnt)
        return ans

    def minCharacters(self, a, b):
        return min(self.go2(a,b),min(self.go1(a,b),self.go1(b,a)))

File: 2021/01/31/jh20s/contest#225/1737.cpp
class Solution {
public:
int go1(string a, string b) {
	int ans = INT_MAX;
	for (int i = 1; i <= 'z'-'a'; i++) {
		int cnt = 0;
		for (int j = 0; j < a.size(); j++) {
			if (a[j] - 'a' >= i)
				cnt++;
		}
		for (int j = 0; j < b.size(); j++) {
			if (b[j] - 'a' < i)
				cnt++;
		}
		ans = min(ans, cnt);
	}
	return ans;
}

int go2(string a, string b) {
	int ans = INT_MAX;
	for (int i = 0; i <= 'z' - 'a'; i++) {
		int cnt = 0;
		for (int j = 0; j < a.size(); j++) {
			if (a[j] - 'a' != i)
				cnt++;
		}
		for (int j = 0; j < b.size(); j++) {
			if (b[j] - 'a' != i)
				cnt++;
		}
		ans = min(ans, cnt);
	}
	return ans;
}

int minCharacters(string a, string b) {
	return min(go2(a, b), min(go1(a, b), go1(b, a)));
}
};